hack/verify-commits.sh

```diff
@@ -1,18 +1,62 @@
 #!/bin/bash
 
+set -e -o pipefail
+
 function finish {
 	if [ -f "$commit_msg_filename" ]; then
 		rm -f $commit_msg_filename
 	fi
 }
 trap finish EXIT
 
+function is_merge_commit {
+	commitid=$1
+	local sha="$1"
+	msha=$(git rev-list -1 --merges ${sha}~1..${sha})
+	[ -z "$msha" ] && return 1
+	return 0
+}
+
+if [[ -z "$UPSTREAM_COMMIT" ]]; then
+	if [[ ! -z "$CI" ]]; then
```

Comment:
Where these two variables are coming from?
Also, should we fail if UPSTREAM_COMMIT is not defined and CI is not defined? In that case, we would not set latest_upstream_commit I think

Reply:
in prow CI=true
the rest (UPSTREAM_COMMIT and UPSTREAM_BRANCH are made especially for users who want to tweak this script)

Reply:
UPSTREAM_COMMIT take precedence over CI (and UPSTREAM_BRANCH). so if we ever have a desire to override it there, we can.
if it's not CI, then UPSTREAM_BRANCH can be used

I'm also defensively checking if branch/ commit exist
as well as that there is a diff

Reply:
Mind adding a comment for that? It's not entirely clear to the new reader